spec/requests/admin/badges_controller_spec.rb

```diff
@@ -244,6 +247,60 @@
 
         expect(response.status).to eq(422)
       end
+
+      context "when grant_existing_holders is true" do
+        it "fails when the badge cannot be granted multiple times" do
+          file = file_from_fixtures('user_emails.csv', 'csv')
+          badge.update!(multiple_grant: false)
+          post "/admin/badges/award/#{badge.id}.json", params: {
+            file: fixture_file_upload(file),
+            grant_existing_holders: true
+          }
+
+          expect(response.status).to eq(422)
+          expect(response.parsed_body['errors']).to eq([
+            I18n.t("badges.mass_award.errors.cant_grant_multiple_times", badge_name: badge.name)
+          ])
+        end
+
+        it "grants the badge to the users in the CSV as many times as they appear in it" do
+          Jobs.run_immediately!
+          user_without_badge = Fabricate(:user)
+          user_with_badge = Fabricate(:user)
+          badge.update!(multiple_grant: true)
+          BadgeGranter.grant(badge, user_with_badge)
+          expect(user_with_badge.reload.badges.pluck(:id)).to eq([badge.id])
+          expect(user_without_badge.reload.badges.pluck(:id)).to eq([])
```

Comment:
```suggestion
```

I don't feel like these assertions are necessary as long as we're certain that we've fabricated correctly